nilearn/decomposition/_base.py

```diff
@@ -0,0 +1,533 @@
+
+        # Mask is provided or computed
+        self.mask = mask
+
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.mask_args = mask_args
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def fit(self, imgs, y=None):
+        """Base fit for decomposition estimators : compute the embedded masker
+
+        Parameters
+        ----------
+        imgs: list of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which the mask is calculated. If this is a list,
+            the affine is considered the same for all.
+        """
+        self.masker_ = check_embedded_nifti_masker(self)
+
+        # Avoid warning with imgs != None if masker_ has been provided a mask_img
+        if self.masker_.mask_img is None:
+            self.masker_.fit(imgs)
+        else:
+            self.masker_.fit()
+        self.mask_img_ = self.masker_.mask_img_
+
+    def _check_components_(self):
+        if not hasattr(self, 'components_'):
+            if self.__class__.__name__ == '_DecompositionEstimator':
+                raise ValueError("Object has no components_ attribute. This may be because"
+                                 "_DecompositionEstimator is direclty being used")
+            else:
+                raise ValueError("Object has no components_ attribute."
+                                 "This is probably because fit has not been called")
+
+    def transform(self, imgs, confounds=None):
+        """ Project the data into a reduced representation
+
+        Parameters
+        ----------
+        imgs: iterable of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data to be projected
+
+        confounds: CSV file path or 2D matrix
+            This parameter is passed to nilearn.signal.clean. Please see the
+            related documentation for details
+        """
+
+        self._check_components_()
+        components_img_ = self.masker_.inverse_transform(self.components_)
+        nifti_maps_masker = NiftiMapsMasker(
+            components_img_, self.masker_.mask_img_,
+            resampling_target='maps')
+        nifti_maps_masker.fit()
+        # XXX: dealing properly with 4D/ list of 4D data?
+        if confounds is None:
+            confounds = itertools.repeat(None, len(imgs))
+        return [nifti_maps_masker.transform(img, confounds=confound)
+                for img, confound in zip(imgs, confounds)]
+
+    def inverse_transform(self, component_signals):
```

Comment:
As discussed earlier with Michael, there is some controvery how to call these things. Many appear to say "component weights" for the actual maps in our case and "loadings" for the coefficients of the linear combination of these maps.

Means: here better ``loadings``or something like that